src/core/thread/mle.cpp

```diff
@@ -1810,6 +1810,8 @@ ThreadError Mle::HandleParentResponse(const Message &aMessage, const Ip6::Messag
     SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kLinkMargin, sizeof(linkMarginTlv), linkMarginTlv));
     VerifyOrExit(linkMarginTlv.IsValid(), error = kThreadError_Parse);
 
+    VerifyOrExit(threadMessageInfo != NULL, error = kThreadError_NotFound; otLogDebgMle("thread message info not found\n"));
```

Comment:
MLE really depends on having link info.  Rather than just dropping the packet, I think the top of `Mle::HandleUdpReceive()` should have an `assert()` to ensure that `aMessageInfo.mLinkInfo != NULL`.